pandas/indexes/base.py

```diff
@@ -86,6 +86,11 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    from pandas.types.generic import ABCPeriodIndex
```

Comment:
Yeah this is a bit of legacy mess. Could do with a clean up. We'll always need `_new_Index` for existing pickles, but with some basic clean up we could remove a bunch of code and this would be redundant for future pickles